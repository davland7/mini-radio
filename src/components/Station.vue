<template>
  <div class="station">
    <div
      aria-pressed="false"
      class="station__button"
      role="button"
      tabindex="0"
      :title="messages.play"
      @click="play"
      @keyup.enter="play"
    >
      <img
        :alt="item.title"
        class="station__logo"
        :src="require(`@/images/${item.logo}`)"
      >
      <svg
        aria-hidden="true"
        class="station__play"
        viewBox="0 0 16 16"
      >
        <path d="M3 2l10 6-10 6z" />
      </svg>
    </div>
    <div class="station__text">
      <strong>{{ item.title }}</strong>
      <span>{{ item.description }}</span>
    </div>
  </div>
</template>

<script>
  export default {
    name: 'StationComponent',
    props: {
      messages: {
        type: Object,
        required: true
      },
      item: {
        type: Object,
        required: true
      }
    },
    emits: [
      'play'
    ],
    methods: {
      play () {
        this.$emit('play', this.item)
      }
    }
  }
</script>

<style lang="scss">
  .station {
    display: flex;
    height: 64px;
    padding: 2px;
    margin: 2px;
    border: solid 1px var(--border-color);
    background-color: var(--content-color);
    border-radius: 60px 0 0 60px;

    &__text {
      display: flex;
      flex-direction: column;
      justify-content: center;
      margin: 0 16px;
      font-size: 14px;
      line-height: 1.5;
      color: var(--light-color);

      strong {
        color: var(--dark-color);
      }
    }

    &__button {
      position: relative;
      width: 60px;
      cursor: pointer;
    }

    &__logo {
      position: absolute;
      width: 58px;
      height: 58px;
      border-radius: 58px;
      border: solid 1px var(--border-color);
    }

    &__play {
      position: absolute;
      width: 100%;
      height: 100%;
      padding: 9px 6px 9px 12px;
      opacity: 0;

      &:hover {
        opacity: .8;
      }
    }
  }
</style>
