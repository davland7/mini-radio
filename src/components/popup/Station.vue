<template>
  <div class="station">
    <div @click="play" :title="messages.play" class="station__button">
      <img :src="station.logo" class="station__logo" />
      <svg viewBox="0 0 16 16" class="station__play">
        <path d="M3 2l10 6-10 6z"></path>
      </svg>
    </div>
    <div class="station__text">
      <strong>{{ station.title }}</strong>
      <span>{{ station.description }}</span>
    </div>
  </div>
</template>

<script>
  export default {
    name: 'Station',
    props: [
      'station'
    ],
    data () {
      return {
        messages: {
          play: chrome.i18n.getMessage('play')
        }
      }
    },
    methods: {
      play () {
        const station = this.station

        this.$emit('play', station)
      }
    }
  }
</script>

<style lang="scss">
  .station {
    display: flex;
    height: 60px;
    padding: 2px;
    margin: 2px;
    border: solid 1px var(--border-color);
    background-color: var(--content-color);

    &__text {
      display: flex;
      flex-direction: column;
      justify-content: space-around;
      width: 242px;
      padding: 0 10px;
      font-size: 14px;
      color: var(--light-color);

      strong {
        color: var(--dark-color);
      }
    }

    &__button {
      position: relative;
      width: 60px;
    }

    &__logo {
      position: absolute;
      width: 60px;
      height: 60px;
    }

    &__play {
      position: absolute;
      width: 40px;
      height: 40px;
      padding: 9px;
      opacity: 0;

      &:hover {
        cursor: pointer;
        opacity: .8;
      }
    }
  }
</style>
