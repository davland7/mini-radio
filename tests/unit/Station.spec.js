import { shallowMount } from '@vue/test-utils'
import Station from '@/components/Station.vue'

const props = {
  messages: {
    play: 'Play'
  },
  item: {
    title: '92.5 The Beat',
    description: 'MontrÃ©al',
    src: '#',
    logo: 'ckbe-fm.png'
  }
}

describe('Station.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(Station, { props })
    expect(wrapper.element).toMatchSnapshot()
  })
})
