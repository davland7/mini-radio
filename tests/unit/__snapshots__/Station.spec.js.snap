// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Station.vue renders correctly 1`] = `
<div
  class="station"
>
  <div
    aria-pressed="false"
    class="station__button"
    role="button"
    tabindex="0"
    title="Play"
  >
    <img
      alt="92.5 The Beat"
      class="station__logo"
      src=""
    />
    <svg
      aria-hidden="true"
      class="station__play"
      viewBox="0 0 16 16"
    >
      <path
        d="M3 2l10 6-10 6z"
      />
    </svg>
  </div>
  <div
    class="station__text"
  >
    <strong>
      92.5 The Beat
    </strong>
    <span>
      Montr√©al
    </span>
  </div>
</div>
`;
m50699Land

m50699Land